src/lib/Form/Type/User/UserModeChoiceType.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\AdminUi\Form\Type\User;
+
+use Ibexa\AdminUi\UserSetting\UserMode;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\ChoiceList\ChoiceList;
+use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+final class UserModeChoiceType extends AbstractType
+{
+    private const TRANSLATION_DOMAIN = 'ibexa_user_settings';
+
+    public function configureOptions(OptionsResolver $resolver): void
+    {
+        $resolver->setDefaults([
+            'choice_loader' => ChoiceList::lazy($this, $this->getChoiceLoader()),
```

Comment:
Don't really see the reasoning behind introducing a Choice Loader here.

`choices` option would be sufficient.

Additionally, you may consider using a `choice_label` option so that forms do not send `'user.setting.mode.expert'` as form field value.